src/App.js

```diff
@@ -13,23 +13,73 @@ import mockCats from './mockCats';
 
 function App() {
 
+  const readThunderCat = () => {
+    fetch("http://localhost:3000/cats")
+    .then((response) => response.json())
+    .then((payload) => setCat(payload))
+    .catch((errors) => console.log("Cat Read ERRORS:", errors))
+  }
+
   const createCat = (cat) => {
-    console.log(cat)
+    fetch("http://localhost:3000/cats", {
+      body: JSON.stringify(cat),
+      headers: {
+        "Content-Type": "application/json"
+      },
+
+      method: "POST"
+    })
+
+    .then((response) => response.json())
+    .then(() => readThunderCat())
+    .catch((errors) => console.log("Cat creat erorrors:", errors))
   }
+  
   const updateCat = (cat, id) => {
-    console.log("cat:", cat)
-    console.log("id:", id)
+    fetch(`http://localhost:3000/cats/${id}`, {
+
+    body: JSON.stringify(cat),
+
+    headers: {
+      "Content-Type": "application/json"
+    },
+
+    method: "PATCH"
+    })
+
+    .then((response) => response.json())
+    .then(() => readThunderCat())
+    .catch((errors) => console.log("Cat update errors:", errors))
   }
 
-  const [cats, setCat] = useState(mockCats)
+  const deleteCat = (id) => {
+    fetch(`http://localhost:3000/cats/${id}`, {
+
+      headers: {
+        "Content-Type": "application/json"
+      },
+
+      method: "DELETE"
+    })
+
+    .then((response) => response.json())
+    .then(() => readThunderCat())
+    .catch((errors) => console.log("Cat delete errors:", errors))
+  }
+
+  const [cats, setCat] = useState([])
+
+  useEffect(() => {
+    readThunderCat()
+  }, [])
```

Comment:
It is most common to have state at the top of the function before any additional functions.  Functionally, this will still work, but placing it at the top of the function is standard practice.